api/jobs.go

```diff
@@ -9,20 +9,34 @@ import (
 
 // Job represents a Buildkite Agent API Job
 type Job struct {
-	ID                 string                `json:"id,omitempty"`
-	Endpoint           string                `json:"endpoint"`
-	State              string                `json:"state,omitempty"`
-	Env                map[string]string     `json:"env,omitempty"`
-	Step               *pipeline.CommandStep `json:"step,omitempty"`
-	ChunksMaxSizeBytes int                   `json:"chunks_max_size_bytes,omitempty"`
-	Token              string                `json:"token,omitempty"`
-	ExitStatus         string                `json:"exit_status,omitempty"`
-	Signal             string                `json:"signal,omitempty"`
-	SignalReason       string                `json:"signal_reason,omitempty"`
-	StartedAt          string                `json:"started_at,omitempty"`
-	FinishedAt         string                `json:"finished_at,omitempty"`
-	RunnableAt         string                `json:"runnable_at,omitempty"`
-	ChunksFailedCount  int                   `json:"chunks_failed_count,omitempty"`
+	ID                 string               `json:"id,omitempty"`
+	Endpoint           string               `json:"endpoint"`
+	State              string               `json:"state,omitempty"`
+	Env                map[string]string    `json:"env,omitempty"`
+	Step               pipeline.CommandStep `json:"step,omitempty"`
+	ChunksMaxSizeBytes int                  `json:"chunks_max_size_bytes,omitempty"`
+	Token              string               `json:"token,omitempty"`
+	ExitStatus         string               `json:"exit_status,omitempty"`
+	Signal             string               `json:"signal,omitempty"`
+	SignalReason       string               `json:"signal_reason,omitempty"`
+	StartedAt          string               `json:"started_at,omitempty"`
+	FinishedAt         string               `json:"finished_at,omitempty"`
+	RunnableAt         string               `json:"runnable_at,omitempty"`
+	ChunksFailedCount  int                  `json:"chunks_failed_count,omitempty"`
+}
+
+func (j Job) ValuesForFields(fields []string) (map[string]string, error) {
```

Comment:
`Job` looks big enough that I'd make this receiver `*Job` (for not-copying-it-over-stack reasons, not for modifying-the-fields reasons).